lib/matplotlib/patches.py

```diff
@@ -1066,10 +1066,18 @@ def set_xy(self, xy):
         ----------
         xy : (N, 2) array-like
             The coordinates of the vertices.
+
+        Notes
+        -----
+        Unlike `~.path.Path`, we do not ignore the last input vertex. If the
+        polygon is meant to be closed, and the last point of the polygon is not
+        equal to the first, we assume that the user has not explicitly passed a
+        ``CLOSEPOLY`` vertex, and add it ourselves.
         """
         xy = np.asarray(xy)
+        N, _ = xy.shape
         if self._closed:
-            if len(xy) and (xy[0] != xy[-1]).any():
+            if N == 1 or N > 1 and (xy[0] != xy[-1]).any():
                 xy = np.concatenate([xy, [xy[0]]])
         else:
             if len(xy) > 2 and (xy[0] == xy[-1]).all():
```

Comment:
Since you have `N` now, I expect you want to use it here too?

Reply:
Clearly I need to get more sleep, fixed.